pkgs/development/python-modules/scalene/default.nix

```diff
@@ -0,0 +1,54 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, setuptools
+, setuptools-scm
+, wheel
+, cloudpickle
+, cython
+, jinja2
+, psutil
+, pynvml
+, rich
+}:
+
+buildPythonPackage rec {
+  pname = "scalene";
+  version = "1.5.35";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-3s0KN06RivCN4G5/LZf+r7GtAlKwqRuZwUtrY2T2Bew=";
+  };
+
+  nativeBuildInputs = [
+    cython
+    setuptools
+    setuptools-scm
+    wheel
+  ];
+
+  propagatedBuildInputs = [
+    cloudpickle
+    jinja2
+    psutil
+    pynvml
+    rich
+    wheel
+  ];
+
+  postPatch = ''
+    substituteInPlace setup.py \
+      --replace-warn "pynvml>=11.0.0,<11.5" "pynvml~=11.5.0"
+    substituteInPlace requirements.txt \
+      --replace-warn "pynvml~=11.0.0" "pynvml~=11.5.0"
+  '';
+
```

Comment:
It would be good to have tests run if possible (note they're often stripped from the PyPI distribution).

Reply:
Agreed. Building from the GitHub repo is tricky though as it clones three additional git repos into the main checked out sources then runs make on them. Making that work in a derivation is a bit beyond my current skills. :)

Reply:
Two of the repos are source-only C/C++ and the third is Cython. There's a small bit of C/C++ code being built via a makefile. 

https://github.com/emeryberger/Heap-Layers
https://github.com/mpaland/printf
https://github.com/plasma-umass/crdp

How does this sound:

1.  Download the first two as source packages.
2.  Install the third into nixpkgs
3.  Patch the Makefile to remove the downloads and use the downloaded source packages

Reply:
I think this is an acceptable approach, although the packages which are simply fetched but not included in top-level/all-packages.nix won't, for example, receive r-ryantm auto-updates, so long term it would be best to build them properly.

Reply:
I'm willing to do the proper thing even if it takes longer. I have scalene working for my own uses, I just think it deserves to be shared.

Are there other source-only libraries in Nixos? `Heap-Layers` and `printf` don't make binaries except for a test function in printf.

Also, thank you for all your thoughtful help.

Reply:
`eigen` is a well-known example of a header-only library (produces no shared object); I'm sure there are plenty of others as well.

Reply:
> Building from the GitHub repo is tricky though as it clones three additional git repos into the main checked out sources then runs make on them. Making that work in a derivation is a bit beyond my current skills.

Please set `meta.sourceProvenance` if you can't figure this out.

Reply:
I'm sorry, nixifying a source build from github isn't going to happen for a while. The build system in the original needs to be refactored and I need the project owners cooperation in that. (He's lovely and helpful, but quite busy.) So we'll have to live with a PIP-sourced build for now.

sourceProvenance is still fromSource, as the PIP package contains all the python and C sources. It just lacks the unit tests.